### PR TITLE
docs(typedoc): add ts docs

### LINKED ISSUE #551: Docs: Add tsdoc
see https://github.com/TypeStrong/typedoc/issues/818

### PR DESCRIPTION
**Pull request by dhruvdutt**
**What kind of change does this PR introduce?**
Docs

**Did you add tests for your changes?**

**If relevant, did you update the documentation?**
Yes

**Summary**
Final piece of the TypeScript migration process. 💯 

**Does this PR introduce a breaking change?**

**Other information**
Closes #551

### CODE REVIEW

**Comment by dhruvdutt** [comment]
@ev1stensberg The issue was because typedoc doesn't work with monorepos out-of-the-box. Notice the config file.

```js
module.exports = {
	name: "webpack-cli",
	mode: "modules",
	out: "docs",
	theme: "default",
	exclude: "*.test.ts",
	"external-modulemap": ".*packages\/(@webpack-cli\/[^\/]+)\/.*"
};
```

File: typedoc.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+module.exports = {
+	name: "webpack-cli",
+	mode: "modules",
+	out: "docs",
+	theme: "default",
+	exclude: "*.test.ts",
```
Comment (ematipico) [inline typedoc.js]: we should exclude also .js files? Or the library just parses .ts files?

Comment (dhruvdutt) [reply typedoc.js]: Yeah, we don't have any tests in ts, so this won't affect anything. I'll clean it up anyways.

File: bin/prompt-command.js [hunk 2]
```diff
@@ -30,21 +30,21 @@ const runCommand = (command, args) => {
 
 module.exports = function promptForInstallation(packages, ...args) {
 	const nameOfPackage = "@webpack-cli/" + packages;
-	let packageIsInstalled = false;
+	let packageIsInstalled = true;
```
Comment (ev1stensberg) [inline bin/prompt-command.js]: hmmmm 🇼🇸

File: bin/prompt-command.js [hunk 3]
```diff
@@ -105,6 +105,6 @@ module.exports = function promptForInstallation(packages, ...args) {
 			}
 		});
 	} else {
-		require(pathForCmd).default(...args); // eslint-disable-line
+		require('../packages/migrate').default(...args); // eslint-disable-line
```
Comment (ev1stensberg) [inline bin/prompt-command.js]: 🤔

File: manual/webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+const ExtractTextPlugin = require("extract-text-webpack-plugin");
```
Comment (ev1stensberg) [inline manual/webpack.config.js]: 👎

File: typedoc.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+	name: "webpack-cli",
+	mode: "modules",
+	out: "docs",
+	theme: "default",
```
Comment (ev1stensberg) [inline typedoc.js]: if the theme is default, can't we discard the prop?

Comment (dhruvdutt) [reply typedoc.js]: Agreed. Cleaned.

File: .gitignore [hunk 6]
```diff
@@ -33,3 +33,6 @@ lerna-debug.log
 
 # Yarn lock file
 yarn.lock
+
+# docs map
```
Comment (ev1stensberg) [inline .gitignore]: more descriptive here please

**Pull request merged**